code/modules/library/admin_only.dm

```diff
@@ -0,0 +1,354 @@
+		message_admins("Proc call lead to hide_book being called, blocked")
+		return
+	if(!check_rights_for(admin, R_BAN))
+		stack_trace("[admin.ckey] tried to hide a book without the required perms")
+		return
+
+	var/log_reason = "([admin.ckey]) hid book #[id][reason ? ": \"[reason]\"" : ""]"
+	log_admin_private(log_reason)
+	log_reason = "\[[SQLtime()]\] round #[GLOB.round_id] [log_reason]"
+
+	var/datum/db_query/query_hide_book = SSdbcore.NewQuery({"
+		UPDATE [format_table_name("library")]
+		SET deleted = 1
+		WHERE id = :id
+	"}, list("id" = id))
+	query_hide_book.warn_execute()
+	qdel(query_hide_book)
+
+	var/datum/db_query/query_update_log = SSdbcore.NewQuery({"
+		INSERT INTO [format_table_name("library_action")] (book, reason, ckey, datetime, action, ip_addr)
+		VALUES (:book, :reason, :ckey, Now(), :action, INET_ATON(:ip_addr))
+	"}, list("book" = id, "reason" = reason, "ckey" = admin.ckey, "action" = BOOK_ADMIN_DELETE, "ip_addr" = admin.address))
+	query_update_log.warn_execute()
+	qdel(query_update_log)
+	library_updated()
+	params_changed = TRUE
+	update_db_info()
+
+/obj/machinery/computer/libraryconsole/admin_only_do_not_map_in_you_fucker/proc/unhide_book(id, reason, client/admin)
+	if(!SSdbcore.Connect())
+		can_connect = FALSE
+		message_admins("Could not connect to the DB")
+		return
+	if(IsAdminAdvancedProcCall())
+		stack_trace("Proc call lead to unhide_book being called, blocked")
+		message_admins("Proc call lead to unhide_book being called, blocked")
+		return
+	if(!check_rights_for(admin, R_BAN))
+		stack_trace("[admin.ckey] tried to unhide a book without the required perms")
+		return
+
+	log_admin_private("([admin.ckey]) unhid book #[id]")
+
+	var/datum/db_query/query_unhide_book = SSdbcore.NewQuery({"
+		UPDATE [format_table_name("library")]
+		SET deleted = NULL
+		WHERE id = :id
+	"}, list("id" = id))
+
+	query_unhide_book.warn_execute()
+	qdel(query_unhide_book)
+
+	var/datum/db_query/query_update_log = SSdbcore.NewQuery({"
+		INSERT INTO [format_table_name("library_action")] (book, reason, ckey, datetime, action, ip_addr)
+		VALUES (:book, :reason, :ckey, Now(), :action, INET_ATON(:ip_addr))
+	"}, list("book" = id, "reason" = reason, "ckey" = admin.ckey, "action" = BOOK_ADMIN_RESTORE, "ip_addr" = admin.address))
+	query_update_log.warn_execute()
+	qdel(query_update_log)
+	library_updated()
+	update_db_info()
+
+/// This mostly exists to document the form of the library_action table, since it doesn't do that good a job on its own
+/datum/book_history_entry
+	/// The id of this logged action
+	var/id
+	/// The book id this log applies to
+	var/book
+	/// The reason this action was enacted
+	var/reason
+	/// The admin who performed the action
+	var/ckey
+	/// The time of the action being performed
+	var/datetime
+	/// The action that occured (BOOK_ADMIN_DELETE, BOOK_ADMIN_RESTORE, and legacy BOOK_ADMIN_REPORT)
+	var/action
+	/// The ip address of the admin who performed the action
+	var/ip_addr
```

Comment:
i think it would be neater if you typed out `ip_address` entirely, letters dont' cost much

Reply:
it's mirrored to the name used in the db, I can change that if you'd like

Reply:
That would be kinder... but it makes sense then. Up to you